# Report of potential types of app for users

We are  company that builds Android and iOS mobile apps. We make our apps available on Google Play and the App Store.

We only build apps that are free to download and install, and our main source of revenue consists of in-app ads. This means our revenue for any given app is mostly influenced by the number of users who use our app — the more users that see and engage with the ads, the better.

*Our goal* for this project is to analyze data to help our developers understand what type of apps are likely to attract more users.

In [66]:
from csv import reader

# Open file App Store Data set
opened_file = open("AppleStore.csv")
read_file = reader(opened_file)
ios = list(read_file)
ios_header = ios[0]
ios = ios[1:]

# Open file Google Play data set
opened_file = open("googleplaystore.csv")
read_file = reader(opened_file)
android = list(read_file)
android_header = android[0]
android = android[1:]

In [67]:
def explore_data(dataset, start, end, rows_and_columns=False):
    dataset_slice = dataset[start:end]
    for row in dataset_slice:
        print(row)
        print("\n") #add a new empty line after each row
    
    if rows_and_columns:
        print('Number of rows:', len(dataset))
        print('Number of columns', len(dataset[0]))

print(ios_header)
print("\n")
explore_data(ios, 0, 4, True)

['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 'Games', '38', '5', '18', '1']


['420009108', 'Temple Run', '65921024', 'USD', '0.0', '1724546', '3842', '4.5', '4.0', '1.6.2', '9+', 'Games', '40', '5', '1', '1']


Number of rows: 7197
Number of columns 16


*There are some column names which are not informative enough. You can follow this [documentation](https://www.kaggle.com/ramamet4/app-store-apple-data-set-10k-apps/home) for more information*

In [68]:
print(android_header)
print("\n")
explore_data(android, 0,7, True)

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['Coloring book moana', 'ART_AND_DESIGN', '3.9', '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


['Pixel Draw - Number Art Coloring Book', 'ART_AND_DESIGN', '4.3', '967', '2.8M', '100,000+', 'Free', '0', 'Eve

In [69]:
print(android[10472])
print("\n")
print(android_header)
print("\n")
print(android[0])

['Life Made WI-Fi Touchscreen Photo Frame', '1.9', '19', '3.0M', '1,000+', 'Free', '0', 'Everyone', '', 'February 11, 2018', '1.0.19', '4.0 and up']


['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


In [70]:
del android[10472]

In [71]:
explore_data(android, 0, 3, True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['Coloring book moana', 'ART_AND_DESIGN', '3.9', '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


Number of rows: 10840
Number of columns 13


*Row 10472 is missing Category, so it was deleted*

In [72]:
for app in android:
    name = app[0]
    if name == "Instagram":
        print(app)

['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577446', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66509917', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']


In [73]:
duplicate_apps = []
unique_apps = []

for app in android:
    name = app[0]
    if name in unique_apps:
        duplicate_apps.append(name)
    else:
        unique_apps.append(name)
        
print("Number of duplicate apps: ", len(duplicate_apps))
print("\n")
print("Number of unique apps: ", len(unique_apps))

Number of duplicate apps:  1181


Number of unique apps:  9659


*There are duplicate rows, I will remove the duplicated ones randomly. They have different number of views, so the entry with highest views will be left, the other will be deleted.*

In [74]:
print("Expected dataset length: ", len(android) - 1181)

Expected dataset length:  9659


In [75]:
reviews_max = {}
for app in android:
    name = app[0]
    n_reviews = float(app[3])
    if name in reviews_max and reviews_max[name] < n_reviews:
        reviews_max[name] = n_reviews
    if name not in reviews_max:
        reviews_max[name] = n_reviews

print("Expected length: ", len(reviews_max))

Expected length:  9659


In [76]:
android_clean = []
already_added = []

for app in android:
    name = app[0]
    n_reviews = float(app[3])
    
    if (n_reviews == reviews_max[name]) and (name not in already_added):
        android_clean.append(app)
        already_added.append(name)
print("After deleting row: ", len(android_clean))

After deleting row:  9659


In [77]:
def is_english(string):
    count = 0
    for character in string:
        if ord(character) > 127:
            count += 1
        
    if count > 3:
        return False
    return True
is_english("Instagram")

True

In [78]:
is_english('爱奇艺PPS -《欢乐颂2》电视剧热播')

False

In [79]:
is_english('Docs To Go™ Free Office Suite')

True

In [80]:
is_english('Instachat 😜')

True

In [81]:
english_ios_apps = []
for app in ios:
    name = app[1]
    if is_english(name) == True:
        english_ios_apps.append(app)

print("English IOS apps: ", len(english_ios_apps))

English IOS apps:  6183


In [82]:
explore_data(ios, 0, 3, True)

['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 'Games', '38', '5', '18', '1']


Number of rows: 7197
Number of columns 16


In [83]:
explore_data(english_ios_apps, 0, 3, True)

['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 'Games', '38', '5', '18', '1']


Number of rows: 6183
Number of columns 16


In [84]:
english_android_apps = []
for app in android_clean:
    name = app[0]
    if is_english(name) == True:
        english_android_apps.append(app)

print(android_header)
print("\n")
explore_data(english_android_apps, 0, 3, True)

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


Number of rows: 9614
Number of columns 13


In [85]:
english_free_android = []
for app in english_android_apps:
    price = app[7]
    if price == "0":
        english_free_android.append(app)

explore_data(english_free_android, 0, 3, True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


Number of rows: 8864
Number of columns 13


In [86]:
print(ios_header)

['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


In [87]:
english_free_ios = []

for app in english_ios_apps:
    price = app[4]
    if price == "0.0":
        english_free_ios.append(app)

explore_data(english_free_ios, 0, 3, True)

['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 'Games', '38', '5', '18', '1']


Number of rows: 3222
Number of columns 16


*As we mentioned in the introduction, our aim is to determine the kinds of apps that are likely to attract more users because our revenue is highly influenced by the number of people using our apps.*

*To minimize risks and overhead, our validation strategy for an app idea is comprised of three steps:*

*Build a minimal Android version of the app, and add it to Google Play.*
*If the app has a good response from users, we develop it further.
If the app is profitable after six months, we build an iOS version of the app and add it to the App Store.*
*Because our end goal is to add the app on both Google Play and the App Store, we need to find app profiles that are successful on both markets. For instance, a profile that works well for both markets might be a productivity app that makes use of gamification.*

*Let's begin the analysis by getting a sense of what are the most common genres for each market. For this, we'll need to build frequency tables for a few columns in our data sets.*

In [88]:
print(ios_header)
print("\n")
print(android_header)

['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


In [89]:
def freq_table(dataset, index):
    table = {}
    total = 0
    
    for app in dataset:
        total += 1
        name = app[index]
        if name in table:
            table[name] +=1
        else:
            table[name] = 1
    table_percentage = {}
    for key in table:
        percentage = (table[key]/total)*100
        table_percentage[key] = percentage

    return table_percentage

freq_table(english_free_ios, 11)

{'Book': 0.4345127250155183,
 'Business': 0.5276225946617008,
 'Catalogs': 0.12414649286157665,
 'Education': 3.662321539416512,
 'Entertainment': 7.883302296710118,
 'Finance': 1.1173184357541899,
 'Food & Drink': 0.8069522036002483,
 'Games': 58.16263190564867,
 'Health & Fitness': 2.0173805090006205,
 'Lifestyle': 1.5828677839851024,
 'Medical': 0.186219739292365,
 'Music': 2.0484171322160147,
 'Navigation': 0.186219739292365,
 'News': 1.3345747982619491,
 'Photo & Video': 4.9658597144630665,
 'Productivity': 1.7380509000620732,
 'Reference': 0.5586592178770949,
 'Shopping': 2.60707635009311,
 'Social Networking': 3.2898820608317814,
 'Sports': 2.1415270018621975,
 'Travel': 1.2414649286157666,
 'Utilities': 2.5139664804469275,
 'Weather': 0.8690254500310366}

In [90]:
def display_table(dataset, index):
    table = freq_table(dataset, index)
    table_display = []
    for key in table:
        key_val_as_tuple = (table[key], key)
        table_display.append(key_val_as_tuple)

    table_sorted = sorted(table_display, reverse = True)
    for entry in table_sorted:
        print(entry[1], ':', entry[0])

In [91]:
display_table(english_free_ios, 11)

Games : 58.16263190564867
Entertainment : 7.883302296710118
Photo & Video : 4.9658597144630665
Education : 3.662321539416512
Social Networking : 3.2898820608317814
Shopping : 2.60707635009311
Utilities : 2.5139664804469275
Sports : 2.1415270018621975
Music : 2.0484171322160147
Health & Fitness : 2.0173805090006205
Productivity : 1.7380509000620732
Lifestyle : 1.5828677839851024
News : 1.3345747982619491
Travel : 1.2414649286157666
Finance : 1.1173184357541899
Weather : 0.8690254500310366
Food & Drink : 0.8069522036002483
Reference : 0.5586592178770949
Business : 0.5276225946617008
Book : 0.4345127250155183
Navigation : 0.186219739292365
Medical : 0.186219739292365
Catalogs : 0.12414649286157665


*In Apple Store dataset:*
    **Game** is the most common genre. **Entertainment** is the runner-up. 
    Most of apps designed for entertainments are popular then apps designed for education. 
    There are lot of apps in one particular genres, but it does not mean that those apps have a lot of users.

In [92]:
display_table(english_free_android, 1)

FAMILY : 18.907942238267147
GAME : 9.724729241877256
TOOLS : 8.461191335740072
BUSINESS : 4.591606498194946
LIFESTYLE : 3.9034296028880866
PRODUCTIVITY : 3.892148014440433
FINANCE : 3.7003610108303246
MEDICAL : 3.531137184115524
SPORTS : 3.395758122743682
PERSONALIZATION : 3.3167870036101084
COMMUNICATION : 3.2378158844765346
HEALTH_AND_FITNESS : 3.0798736462093865
PHOTOGRAPHY : 2.944494584837545
NEWS_AND_MAGAZINES : 2.7978339350180503
SOCIAL : 2.6624548736462095
TRAVEL_AND_LOCAL : 2.33528880866426
SHOPPING : 2.2450361010830324
BOOKS_AND_REFERENCE : 2.1435018050541514
DATING : 1.861462093862816
VIDEO_PLAYERS : 1.7937725631768955
MAPS_AND_NAVIGATION : 1.3989169675090252
FOOD_AND_DRINK : 1.2409747292418771
EDUCATION : 1.1620036101083033
ENTERTAINMENT : 0.9589350180505415
LIBRARIES_AND_DEMO : 0.9363718411552346
AUTO_AND_VEHICLES : 0.9250902527075812
HOUSE_AND_HOME : 0.8235559566787004
WEATHER : 0.8009927797833934
EVENTS : 0.7107400722021661
PARENTING : 0.6543321299638989
ART_AND_DESIGN : 

*In Android Store App*: The most common genres are **Family**, **Game**, **Tools** and **Business**. 
    On Android system, apart from Games with the highest frequencies, most of following apps belongs to educations and lifestyle. 

In [93]:
print(ios_header)

['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


In [94]:
print(android_header)

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


In [95]:
explore_data(english_free_android, 0 , 3, True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


Number of rows: 8864
Number of columns 13


In [96]:
explore_data(english_free_ios, 0, 3, True)

['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 'Games', '38', '5', '18', '1']


Number of rows: 3222
Number of columns 16


In [97]:
freq_table(english_free_ios, -5)

{'Book': 0.4345127250155183,
 'Business': 0.5276225946617008,
 'Catalogs': 0.12414649286157665,
 'Education': 3.662321539416512,
 'Entertainment': 7.883302296710118,
 'Finance': 1.1173184357541899,
 'Food & Drink': 0.8069522036002483,
 'Games': 58.16263190564867,
 'Health & Fitness': 2.0173805090006205,
 'Lifestyle': 1.5828677839851024,
 'Medical': 0.186219739292365,
 'Music': 2.0484171322160147,
 'Navigation': 0.186219739292365,
 'News': 1.3345747982619491,
 'Photo & Video': 4.9658597144630665,
 'Productivity': 1.7380509000620732,
 'Reference': 0.5586592178770949,
 'Shopping': 2.60707635009311,
 'Social Networking': 3.2898820608317814,
 'Sports': 2.1415270018621975,
 'Travel': 1.2414649286157666,
 'Utilities': 2.5139664804469275,
 'Weather': 0.8690254500310366}

In [98]:
genre_table = freq_table(english_free_ios, -5)

In [99]:
genre_table = freq_table(english_free_ios, -5)

for genre in genre_table:
    total = 0
    len_genre = 0
    for app in english_free_ios:
        genre_app = app[-5]
        if genre_app == genre:
            value = float(app[5])
            total += value
            len_genre += 1
    average = total / len_genre
    print(genre, ": ", average)

Finance :  31467.944444444445
Lifestyle :  16485.764705882353
Reference :  74942.11111111111
Sports :  23008.898550724636
Entertainment :  14029.830708661417
Shopping :  26919.690476190477
Education :  7003.983050847458
Catalogs :  4004.0
Travel :  28243.8
Utilities :  18684.456790123455
Medical :  612.0
Food & Drink :  33333.92307692308
Health & Fitness :  23298.015384615384
Navigation :  86090.33333333333
Music :  57326.530303030304
Productivity :  21028.410714285714
Social Networking :  71548.34905660378
Games :  22788.6696905016
Photo & Video :  28441.54375
Book :  39758.5
Business :  7491.117647058823
Weather :  52279.892857142855
News :  21248.023255813954


In [100]:
for app in english_free_ios:
    if app[-5] == "Navigation":
        print(app[1], ": ", app[5])

Waze - GPS Navigation, Maps & Real-time Traffic :  345046
Google Maps - Navigation & Transit :  154911
Geocaching® :  12811
CoPilot GPS – Car Navigation & Offline Maps :  3582
ImmobilienScout24: Real Estate Search in Germany :  187
Railway Route Search :  5


In [101]:
for app in english_free_ios:
    if app[-5] == "Social Networking":
        print(app[1], ": ", app[5])

Facebook :  2974676
Pinterest :  1061624
Skype for iPhone :  373519
Messenger :  351466
Tumblr :  334293
WhatsApp Messenger :  287589
Kik :  260965
ooVoo – Free Video Call, Text and Voice :  177501
TextNow - Unlimited Text + Calls :  164963
Viber Messenger – Text & Call :  164249
Followers - Social Analytics For Instagram :  112778
MeetMe - Chat and Meet New People :  97072
We Heart It - Fashion, wallpapers, quotes, tattoos :  90414
InsTrack for Instagram - Analytics Plus More :  85535
Tango - Free Video Call, Voice and Chat :  75412
LinkedIn :  71856
Match™ - #1 Dating App. :  60659
Skype for iPad :  60163
POF - Best Dating App for Conversations :  52642
Timehop :  49510
Find My Family, Friends & iPhone - Life360 Locator :  43877
Whisper - Share, Express, Meet :  39819
Hangouts :  36404
LINE PLAY - Your Avatar World :  34677
WeChat :  34584
Badoo - Meet New People, Chat, Socialize. :  34428
Followers + for Instagram - Follower Analytics :  28633
GroupMe :  28260
Marco Polo Video Walki

In [102]:
for app in english_free_ios:
    if app[-5] == "Reference":
        print(app[1], " ", app[5])

Bible   985920
Dictionary.com Dictionary & Thesaurus   200047
Dictionary.com Dictionary & Thesaurus for iPad   54175
Google Translate   26786
Muslim Pro: Ramadan 2017 Prayer Times, Azan, Quran   18418
New Furniture Mods - Pocket Wiki & Game Tools for Minecraft PC Edition   17588
Merriam-Webster Dictionary   16849
Night Sky   12122
City Maps for Minecraft PE - The Best Maps for Minecraft Pocket Edition (MCPE)   8535
LUCKY BLOCK MOD ™ for Minecraft PC Edition - The Best Pocket Wiki & Mods Installer Tools   4693
GUNS MODS for Minecraft PC Edition - Mods Tools   1497
Guides for Pokémon GO - Pokemon GO News and Cheats   826
WWDC   762
Horror Maps for Minecraft PE - Download The Scariest Maps for Minecraft Pocket Edition (MCPE) Free   718
VPN Express   14
Real Bike Traffic Rider Virtual Reality Glasses   8
教えて!goo   0
Jishokun-Japanese English Dictionary & Translator   0


In [103]:
for app in english_free_ios:
    if app[-5] == "Weather":
        print(app[1], ": ", app[5])

The Weather Channel: Forecast, Radar & Alerts :  495626
The Weather Channel App for iPad – best local forecast, radar map, and storm tracking :  208648
WeatherBug - Local Weather, Radar, Maps, Alerts :  188583
MyRadar NOAA Weather Radar Forecast :  150158
AccuWeather - Weather for Life :  144214
Yahoo Weather :  112603
Weather Underground: Custom Forecast & Local Radar :  49192
NOAA Weather Radar - Weather Forecast & HD Radar :  45696
Weather Live Free - Weather Forecast & Alerts :  35702
Storm Radar :  22792
QuakeFeed Earthquake Map, Alerts, and News :  6081
Moji Weather - Free Weather Forecast :  2333
Hurricane by American Red Cross :  1158
Forecast Bar :  375
Hurricane Tracker WESH 2 Orlando, Central Florida :  203
FEMA :  128
iWeather - World weather forecast :  80
Weather - Radar - Storm with Morecast App :  78
Yurekuru Call :  53
Weather & Radar :  37
WRAL Weather Alert :  25
Météo-France :  24
JaxReady :  22
Freddy the Frogcaster's Weather Station :  14
Almanac Long-Range Weathe

In [104]:
for app in english_free_ios:
    if app[-5] == "Music":
        print(app[1], ": ", app[5])

Pandora - Music & Radio :  1126879
Spotify Music :  878563
Shazam - Discover music, artists, videos & lyrics :  402925
iHeartRadio – Free Music & Radio Stations :  293228
SoundCloud - Music & Audio :  135744
Magic Piano by Smule :  131695
Smule Sing! :  119316
TuneIn Radio - MLB NBA Audiobooks Podcasts Music :  110420
Amazon Music :  106235
SoundHound Song Search & Music Player :  82602
Sonos Controller :  48905
Bandsintown Concerts :  30845
Karaoke - Sing Karaoke, Unlimited Songs! :  28606
My Mixtapez Music :  26286
Sing Karaoke Songs Unlimited with StarMaker :  26227
Ringtones for iPhone & Ringtone Maker :  25403
Musi - Unlimited Music For YouTube :  25193
AutoRap by Smule :  18202
Spinrilla - Mixtapes For Free :  15053
Napster - Top Music & Radio :  14268
edjing Mix:DJ turntable to remix and scratch music :  13580
Free Music - MP3 Streamer & Playlist Manager Pro :  13443
Free Piano app by Yokee :  13016
Google Play Music :  10118
Certified Mixtapes - Hip Hop Albums & Mixtapes :  997

*On average, **navigation** apps have the highest number of user reviews, but this figure is heavily influenced by Waze and Google Maps, which have close to half a million user reviews together*.

*The same pattern applies to social networking apps, where the average number is heavily influenced by a few giants like Facebook, Pinterest, Skype, etc. Same applies to music apps, where a few big players like Pandora, Spotify, and Shazam heavily influence the average number.*

*Our aim is to find popular genres, but navigation, social networking or music apps might seem more popular than they really are. The average number of ratings seem to be skewed by very few apps which have hundreds of thousands of user ratings, while the other apps may struggle to get past the 10,000 threshold. We could get a better picture by removing these extremely popular apps for each genre and then rework the averages, but we'll leave this level of detail for later.*

*Reference apps have 74,942 user ratings on average, but it's actually the Bible and Dictionary.com which skew up the average rating*

*However, this niche seems to show some potential. One thing we could do is take another popular book and turn it into an app where we could add different features besides the raw version of the book. This might include daily quotes from the book, an audio version of the book, quizzes about the book, etc. On top of that, we could also embed a dictionary within the app, so users don't need to exit our app to look up words in an external app.*

*This idea seems to fit well with the fact that the App Store is dominated by for-fun apps. This suggests the market might be a bit saturated with for-fun apps, which means a practical app might have more of a chance to stand out among the huge number of apps on the App Store.*

*Other genres that seem popular include weather, book, food and drink, or finance. The book genre seem to overlap a bit with the app idea we described above, but the other genres don't seem too interesting to us*

*Weather apps — people generally don't spend too much time in-app, and the chances of making profit from in-app adds are low. Also, getting reliable live weather data may require us to connect our apps to non-free APIs.*

*Food and drink — examples here include Starbucks, Dunkin' Donuts, McDonald's, etc. So making a popular food and drink app requires actual cooking and a delivery service, which is outside the scope of our company.*

*Finance apps — these apps involve banking, paying bills, money transfer, etc. Building a finance app requires domain knowledge, and we don't want to hire a finance expert just to build an app.*

In [105]:
print(android_header)

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


In [106]:
explore_data(english_free_android, 0 , 2, True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


Number of rows: 8864
Number of columns 13


In [107]:
type(english_free_android[5])

list

In [108]:
type(english_free_android[5][5])

str

In [109]:
table = freq_table(english_free_android, 1)

In [110]:
print(table)

{'FOOD_AND_DRINK': 1.2409747292418771, 'NEWS_AND_MAGAZINES': 2.7978339350180503, 'BEAUTY': 0.5979241877256317, 'VIDEO_PLAYERS': 1.7937725631768955, 'ART_AND_DESIGN': 0.6430505415162455, 'LIBRARIES_AND_DEMO': 0.9363718411552346, 'EDUCATION': 1.1620036101083033, 'MEDICAL': 3.531137184115524, 'ENTERTAINMENT': 0.9589350180505415, 'DATING': 1.861462093862816, 'SPORTS': 3.395758122743682, 'FINANCE': 3.7003610108303246, 'PARENTING': 0.6543321299638989, 'LIFESTYLE': 3.9034296028880866, 'WEATHER': 0.8009927797833934, 'MAPS_AND_NAVIGATION': 1.3989169675090252, 'SOCIAL': 2.6624548736462095, 'BOOKS_AND_REFERENCE': 2.1435018050541514, 'FAMILY': 18.907942238267147, 'GAME': 9.724729241877256, 'SHOPPING': 2.2450361010830324, 'TOOLS': 8.461191335740072, 'COMMUNICATION': 3.2378158844765346, 'HOUSE_AND_HOME': 0.8235559566787004, 'EVENTS': 0.7107400722021661, 'COMICS': 0.6204873646209386, 'AUTO_AND_VEHICLES': 0.9250902527075812, 'PRODUCTIVITY': 3.892148014440433, 'BUSINESS': 4.591606498194946, 'PERSONALIZ

In [111]:
for category in table:
    total = 0
    len_category = 0
    for app in english_free_android:
        category_app = app[1]
        if category_app == category:
            n_install = app[5]
            n_install = n_install.replace(",", "")
            n_install = n_install.replace("+", "")
            n_install = float(n_install)
            total += n_install
            len_category += 1
    average = total/len_category
    print(category, ": ", average)

FOOD_AND_DRINK :  1924897.7363636363
NEWS_AND_MAGAZINES :  9549178.467741935
BEAUTY :  513151.88679245283
VIDEO_PLAYERS :  24727872.452830188
ART_AND_DESIGN :  1986335.0877192982
LIBRARIES_AND_DEMO :  638503.734939759
EDUCATION :  1833495.145631068
MEDICAL :  120550.61980830671
ENTERTAINMENT :  11640705.88235294
DATING :  854028.8303030303
SPORTS :  3638640.1428571427
FINANCE :  1387692.475609756
PARENTING :  542603.6206896552
LIFESTYLE :  1437816.2687861272
WEATHER :  5074486.197183099
MAPS_AND_NAVIGATION :  4056941.7741935486
SOCIAL :  23253652.127118643
BOOKS_AND_REFERENCE :  8767811.894736841
FAMILY :  3695641.8198090694
GAME :  15588015.603248259
SHOPPING :  7036877.311557789
TOOLS :  10801391.298666667
COMMUNICATION :  38456119.167247385
HOUSE_AND_HOME :  1331540.5616438356
EVENTS :  253542.22222222222
COMICS :  817657.2727272727
AUTO_AND_VEHICLES :  647317.8170731707
PRODUCTIVITY :  16787331.344927534
BUSINESS :  1712290.1474201474
PERSONALIZATION :  5201482.6122448975
HEALTH_AN

On average, **communication** apps have the most installs: 38,456,119. This number is heavily skewed up by a few apps that have over one billion installs (WhatsApp, Facebook Messenger, Skype, Google Chrome, Gmail, and Hangouts), and a few others with over 100 and 500 million installs:

In [112]:
for app in english_free_android:
    if app[1] == "COMMUNICATION" and (app[5] == "1,000,000,000+" or app[5] == "500,000,000+" or app[5] == "100,000,000+"):
        print(app[0],": ", app[5])
        

WhatsApp Messenger :  1,000,000,000+
imo beta free calls and text :  100,000,000+
Android Messages :  100,000,000+
Google Duo - High Quality Video Calls :  500,000,000+
Messenger – Text and Video Chat for Free :  1,000,000,000+
imo free video calls and chat :  500,000,000+
Skype - free IM & video calls :  1,000,000,000+
Who :  100,000,000+
GO SMS Pro - Messenger, Free Themes, Emoji :  100,000,000+
LINE: Free Calls & Messages :  500,000,000+
Google Chrome: Fast & Secure :  1,000,000,000+
Firefox Browser fast & private :  100,000,000+
UC Browser - Fast Download Private & Secure :  500,000,000+
Gmail :  1,000,000,000+
Hangouts :  1,000,000,000+
Messenger Lite: Free Calls & Messages :  100,000,000+
Kik :  100,000,000+
KakaoTalk: Free Calls & Text :  100,000,000+
Opera Mini - fast web browser :  100,000,000+
Opera Browser: Fast and Secure :  100,000,000+
Telegram :  100,000,000+
Truecaller: Caller ID, SMS spam blocking & Dialer :  100,000,000+
UC Browser Mini -Tiny Fast Private & Secure :  

If we removed all the communication apps that have over 100 million installs, the average would be reduced roughly ten times:

In [113]:
under_10 = []

for app in english_free_android:
    value = app[5]
    value = value.replace(",","")
    value = value.replace("+", "")
    value = float(value)
    if app[1] == "COMMUNICATION" and value < 100000000:
        under_10.append(value)
sum(under_10)/len(under_10) 

3603485.3884615386

We see the same pattern for the video players category, which is the runner-up with 24,727,872 installs. The market is dominated by apps like Youtube, Google Play Movies & TV, or MX Player. The pattern is repeated for social apps (where we have giants like Facebook, Instagram, Google+, etc.), photography apps (Google Photos and other popular photo editors), or productivity apps (Microsoft Word, Dropbox, Google Calendar, Evernote, etc.).

Again, the main concern is that these app genres might seem more popular than they really are. Moreover, these niches seem to be dominated by a few giants who are hard to compete against.

The game genre seems pretty popular, but previously we found out this part of the market seems a bit saturated, so we'd like to come up with a different app recommendation if possible.

The books and reference genre looks fairly popular as well, with an average number of installs of 8,767,811. It's interesting to explore this in more depth, since we found this genre has some potential to work well on the App Store, and our aim is to recommend an app genre that shows potential for being profitable on both the App Store and Google Play.

Let's take a look at some of the apps from this genre and their number of installs:

In [114]:
for app in english_free_android:
    print(app[0], ": ", app[5])

Photo Editor & Candy Camera & Grid & ScrapBook :  10,000+
U Launcher Lite – FREE Live Cool Themes, Hide Apps :  5,000,000+
Sketch - Draw & Paint :  50,000,000+
Pixel Draw - Number Art Coloring Book :  100,000+
Paper flowers instructions :  50,000+
Smoke Effect Photo Maker - Smoke Editor :  50,000+
Infinite Painter :  1,000,000+
Garden Coloring Book :  1,000,000+
Kids Paint Free - Drawing Fun :  10,000+
Text on Photo - Fonteee :  1,000,000+
Name Art Photo Editor - Focus n Filters :  1,000,000+
Tattoo Name On My Photo Editor :  10,000,000+
Mandala Coloring Book :  100,000+
3D Color Pixel by Number - Sandbox Art Coloring :  100,000+
Learn To Draw Kawaii Characters :  5,000+
Photo Designer - Write your name with shapes :  500,000+
350 Diy Room Decor Ideas :  10,000+
FlipaClip - Cartoon animation :  5,000,000+
ibis Paint X :  10,000,000+
Logo Maker - Small Business :  100,000+
Boys Photo Editor - Six Pack & Men's Suit :  100,000+
Superheroes Wallpapers | 4K Backgrounds :  500,000+
HD Mickey

The book and reference genre includes a variety of apps: software for processing and reading ebooks, various collections of libraries, dictionaries, tutorials on programming or languages, etc. It seems there's still a small number of extremely popular apps that skew the average:

In [117]:
for app in english_free_android:
    if app[1] == "BOOKS_AND_REFERENCE" and (app[5] == "1,000,000,000+" or app[5] == "500,000,000+" or app[5] == "100,000,000+"):
        print(app[0],": ", app[5])
        

Google Play Books :  1,000,000,000+
Bible :  100,000,000+
Amazon Kindle :  100,000,000+
Wattpad 📖 Free Books :  100,000,000+
Audiobooks from Audible :  100,000,000+


However, it looks like there are only a few very popular apps, so this market still shows potential. Let's try to get some app ideas based on the kind of apps that are somewhere in the middle in terms of popularity (between 1,000,000 and 100,000,000 downloads):

In [118]:
for app in english_free_android:
    if app[1] == "BOOKS_AND_REFERENCE" and (app[5] == "1,000,000+" or app[5] == "5,000,000+" or app[5] == "10,000,000+" or app[5] == "100,000,000+"):
        print(app[0], ": ", app[5])

Wikipedia :  10,000,000+
Cool Reader :  10,000,000+
Book store :  1,000,000+
FBReader: Favorite Book Reader :  10,000,000+
Free Books - Spirit Fanfiction and Stories :  1,000,000+
AlReader -any text book reader :  5,000,000+
FamilySearch Tree :  1,000,000+
Cloud of Books :  1,000,000+
ReadEra – free ebook reader :  1,000,000+
Ebook Reader :  5,000,000+
Read books online :  5,000,000+
eBoox: book reader fb2 epub zip :  1,000,000+
All Maths Formulas :  1,000,000+
Ancestry :  5,000,000+
HTC Help :  10,000,000+
Moon+ Reader :  10,000,000+
English-Myanmar Dictionary :  1,000,000+
Golden Dictionary (EN-AR) :  1,000,000+
All Language Translator Free :  1,000,000+
Bible :  100,000,000+
Amazon Kindle :  100,000,000+
Aldiko Book Reader :  10,000,000+
Wattpad 📖 Free Books :  100,000,000+
Dictionary - WordWeb :  5,000,000+
50000 Free eBooks & Free AudioBooks :  5,000,000+
Al-Quran (Free) :  10,000,000+
Al Quran Indonesia :  10,000,000+
Al'Quran Bahasa Indonesia :  10,000,000+
Al Quran Al karim :  

This niche seems to be dominated by software for processing and reading ebooks, as well as various collections of libraries and dictionaries, so it's probably not a good idea to build similar apps since there'll be some significant competition.

We also notice there are quite a few apps built around the book Quran, which suggests that building an app around a popular book can be profitable. It seems that taking a popular book (perhaps a more recent book) and turning it into an app could be profitable for both the Google Play and the App Store markets.

However, it looks like the market is already full of libraries, so we need to add some special features besides the raw version of the book. This might include daily quotes from the book, an audio version of the book, quizzes on the book, a forum where people can discuss the book, etc.